packages/rspack-cli/src/utils/buildConfig.ts

```diff
@@ -0,0 +1,118 @@
+import { Mode, RspackOptions } from "@rspack/core";
+import { RspackCLIOptions } from "../types";
+
+export async function buildConfigWithOptions(
+	item: RspackOptions,
+	options: RspackCLIOptions,
+	isColorSupported: boolean
+) {
+	const mode = buildConfig_mode(item, options);
+	buildConfig_stats(item, isColorSupported);
+	buildConfig_analyze(item, options);
+	buildConfig_builtin(item, mode);
+	buildConfig_devtool(item, mode);
+	buildConfig_watch(item, options);
+}
+
+function buildConfig_mode(
+	item: RspackOptions,
+	options: RspackCLIOptions
+): Mode {
+	// Respect `process.env.NODE_ENV`
+	if (
+		!item.mode &&
+		process.env &&
+		process.env.NODE_ENV &&
+		(process.env.NODE_ENV === "development" ||
+			process.env.NODE_ENV === "production" ||
+			process.env.NODE_ENV === "none")
+	) {
+		item.mode = process.env.NODE_ENV;
+	}
```

Comment:
```suggestion
  if (!item.mode && process.env && ['development', 'production', 'none'].includes(process.env.NODE_ENV)) {
   item.mode = process.env.NODE_ENV;
  }
```

Reply:
That way it allows us to add more ENVs in _future_ like `ci`, as well as some standard aliases if any, etc.; _increasing_ readability and _reducing_ verboseness for the exactly same task. :)

Reply:
😂Thank you for your suggestion. I just copied this code from webpack-cli